test_util/test_usage.py

```diff
@@ -25,19 +25,44 @@ def trivial_env(monkeypatch):
     monkeypatch.setenv('PUBLIC_MASTER_HOSTS', '127.0.0.1,0.0.0.0')
     monkeypatch.setenv('SLAVE_HOSTS', '127.0.0.1,0.0.0.0')
     monkeypatch.setenv('PUBLIC_SLAVE_HOSTS', '127.0.0.1,0.0.0.0')
-    monkeypatch.setenv('DNS_SEARCH', 'false')
-    monkeypatch.setenv('DCOS_PROVIDER', 'onprem')
 
 
 def test_make_user_session(monkeypatch, trivial_env):
-    monkeypatch.setattr(ClusterApi, 'post',
-                        lambda *args, **kwargs: MockResponse({'token': 'abc'}, {'dcos-acs-auth-cookie': 'foo'}))
+    monkeypatch.setattr(requests, 'post', lambda *args, **kwargs: MockResponse())
+    monkeypatch.setattr(DcosApiSession, 'wait_for_dcos', lambda self: True)
     user_1 = DcosUser({'foo': 'bar'})
     user_2 = DcosUser({'baz': 'qux'})
     args = get_args_from_env()
-    args['web_auth_default_user'] = user_1
-    cluster_none = ClusterApi(**args)
+    args['auth_user'] = user_1
+    cluster_none = DcosApiSession(**args)
+    # make user session from no auth
     cluster_1 = cluster_none.get_user_session(user_1)
+    assert cluster_1.session.auth.auth_token == 'bar'
+    # Add a cookie to this session to make sure it gets cleared
+    cluster_1.session.cookies.update({'dcos-acs-auth-cookie': 'foo'})
+    # make user session from user
     cluster_2 = cluster_1.get_user_session(user_2)
-    assert cluster_1.default_headers['Authorization'] == 'token=abc'
-    assert cluster_2.default_headers['Authorization'] == 'token=abc'
+    assert cluster_2.session.auth.auth_token == 'bar'
+    # check cleared cookie
+    assert cluster_2.session.cookies.get('dcos-acs-auth-cookie') is None
+    # make no auth session from use session
+    cluster_none = cluster_2.get_user_session(None)
+    assert cluster_none.session.auth is None
+    assert len(cluster_none.session.cookies.items()) == 0
+
+
+class MyTestClient(ApiClientSession):
```

Comment:
Is the purpose of this test is to exercise 'super().api_request`?

1. This relies on external network. is it a good idea to do this without the network requirement? ( Just run a simple HTTP server class in the thread).

2. The point of 404 and then 200 is not obvious at all. Why do this in a test?